server/src/internalClusterTest/java/org/elasticsearch/cluster/service/ClusterServiceIT.java

```diff
@@ -441,4 +439,28 @@ public void onFailure(Exception e) {
             block2.countDown();
         }
     }
+
+    private static void waitForTimeToElapse() throws InterruptedException {
+        final ThreadPool[] threadPools = StreamSupport.stream(internalCluster().getInstances(ClusterService.class).spliterator(), false)
+            .map(ClusterService::threadPool)
+            .toArray(ThreadPool[]::new);
+        final long[] startTimes = Arrays.stream(threadPools).mapToLong(ThreadPool::relativeTimeInMillis).toArray();
+
+        final var startNanoTime = System.nanoTime();
+        while (TimeUnit.MILLISECONDS.convert(System.nanoTime() - startNanoTime, TimeUnit.NANOSECONDS) <= 100) {
+            // noinspection BusyWait
+            Thread.sleep(100);
+        }
+
+        outer: do {
+            for (int i = 0; i < threadPools.length; i++) {
+                if (threadPools[i].relativeTimeInMillis() <= startTimes[i]) {
```

Comment:
Is it even possible that the `relativeTimeInMillis()` for a threadpool could become less than when you recorded its start time above?

Reply:
It certainly shouldn't go backwards, no. But that doesn't really matter here, and IMO it'd be kind of weird to say `threadPools[i].relativeTimeInMillis() == startTimes[I]` in this condition here.